hdfs3/core.py

```diff
@@ -694,8 +694,12 @@ def write(self, data):
             return
         if not _lib.hdfsFileIsOpenForWrite(self._handle):
             raise IOError('File not write mode')
-        if not _lib.hdfsWrite(self._fs, self._handle, data, len(data)) == len(data):
-            raise IOError('Write failed on file %s' % self.path)
+        write_block = 64 * 2**20
+        for offset in range(0, len(data), write_block):
```

Comment:
Might want to test that this works when the blocksize doesn't evenly divide the data